File: decode.cpp
#include <fstream>
#include <iostream>
int main(int argc, char** argv){
    std::ifstream in(argv[1], std::ios::binary);
    std::ofstream out(argv[2], std::ios::binary);
    char nontpcounter = 0;
    while(true){
        unsigned char i = in.get();
        std::cout<<(int)i<<" ";
        if(in.eof()) break;
        if(nontpcounter>0){out<<i;nontpcounter--;continue;}
        if(i>0xaf) {nontpcounter=i-0xaf;continue;}
        out<<(char)0xf3<<(char)0xb1<<(char)(0xa4+((i&0xc0)>>6))<<(char)((i&0x3f)+128);
    }
}

File: decode.py
from sys import argv
with open(input("Input file: ") if len(argv)<2 else argv[1],"rb") as f:
    with open(input("Output file: ") if len(argv)<3 else argv[2],"wb") as out:
        nontpcounter = 0
        for i in f.read():
            if nontpcounter>0: 
                out.write(i.to_bytes(1))
                nontpcounter-=1
                continue
            if i>0xaf:
                nontpcounter=i-0xaf
                continue
            out.write(b"\xf3\xb1")
            out.write((0xa4+((i&0xc0)>>6)).to_bytes(1))
            out.write(((i&0x3f)+128).to_bytes(1))